packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -279,21 +279,20 @@ export const chainflipApi: SwapperApi = {
   checkTradeStatus: async ({
     config,
     quoteId,
+    swap,
   }): Promise<{
     status: TxStatus
     buyTxHash: string | undefined
     message: string | [string, InterpolationOptions] | undefined
   }> => {
-    const swap = tradeQuoteMetadata.get(quoteId)
-    if (!swap) throw Error(`Missing trade quote metadata for quoteId ${quoteId}`)
-    // Note, the swapId isn't the quoteId - we set the swapId at pre-execution time, when getting the receive addy and instantiating a flip swap
-    const swapId = swap.id
+    const chainflipSwapId = tradeQuoteMetadata.get(quoteId)?.id || swap?.metadata.chainflipSwapId
```

Comment:
Beautiful.

Very second line I'm reading in this PR, but in the spirit of consuming the groundwork done here, shouldn't we *only* rely on swap for the swap-specific metadata?
i.e two birds one stone, not saying we ditch its upsertion altogether (we could cleanup trade quote types later) but would make sense to not duplicate source of truths, consume the groundwork done here, and while at that, tackle https://github.com/shapeshift/web/issues/9606 partially 🎉

Reply:
In effect, that means for this particular line:
```suggestion
    const chainflipSwapId = swap?.metadata.chainflipSwapId
```

Reply:
Did it in [fbcd366](https://github.com/shapeshift/web/pull/9625/commits/fbcd366ded019ee7c8586022801e2a9625cc80dd)

Also needed to change something on relay side